docs: revised the application progress bar feature page (#26003)
electron/electron: docs/tutorial/progress-bar.md

# Progress Bar in Taskbar (Windows, macOS, Unity)

> ## Overview

Reviewer: Can we add a simple paragraph about what this feature does and then segue into the platform-specific details?

> On Windows, a taskbar button can be used to display a progress bar. This enables

Reviewer: ```suggestion
On Windows, you can use a taskbar button to display a progress bar. This enables
```

I feel like we're using "You" a lot across the different guides, so consistent tone is probably useful here.

a window to provide progress information to the user without the need of switching to
the window itself.

![Taskbar Progress Bar][taskbar-progress-image]

On macOS, the progress bar will be displayed as a part of the dock icon.

> The Unity DE also has a similar feature that allows you to specify the progress

Reviewer: Would be helpful to expand "Unity DE" to users who are unfamiliar with Linux-specific terms.
```suggestion
On Linux, the Unity graphical interface also has a similar feature that allows you to specify the progress
```

bar in the launcher.

----

All three cases are covered by the same API - the
[`setProgressBar()`][setprogressbar] method available on an instance of
`BrowserWindow`. To indicate your progress, call this method with a number
between `0` and `1`. For example, if you have a long-running task that is
currently at 63% towards completion, you would call it as
`setProgressBar(0.63)`.

Setting the parameter to negative values (e.g. `-1`) will remove the progress
bar, whereas setting it to values greater than `1` (e.g. `2`) will switch the

> progress bar to intermediate mode (100% complete).

Reviewer: I think this was supposed to say "indeterminate" mode, but the typo persisted for a long time without anyone noticing.

See API doc: https://www.electronjs.org/docs/api/browser-window#:~:text=Change%20to%20indeterminate%20mode%20when%20progress%20%3E%201

Author: Exactly. API is clear on that

See the [API documentation for more options and modes][setprogressbar].

## Example

Starting with a working application from the
[Quick Start Guide](quick-start.md), add the following lines to the
`main.js` file:

```javascript
const { BrowserWindow } = require('electron')
const win = new BrowserWindow()

win.setProgressBar(0.5)
```

After launching the Electron application, you should see the bar in
the dock (macOS) or taskbar (Windows, Unity), indicating the progress
percentage you just defined.

![macOS dock progress bar](../images/dock-progress-bar.png)

For macOS, the progress bar will also be indicated for your application

> when using [Mission control](https://support.apple.com/en-us/HT204100):

Reviewer: ```suggestion
when using [Mission Control](https://support.apple.com/en-us/HT204100):
```

# Progress Bar in Taskbar (Windows, macOS, Unity)

## Overview

A progress bar enables a window to provide progress information to the user
without the need of switching to the window itself.

On Windows, you can use a taskbar button to display a progress bar.

![Windows Progress Bar][windows-progress-bar]

On macOS, the progress bar will be displayed as a part of the dock icon.

![macOS Progress Bar][macos-progress-bar]

On Linux, the Unity graphical interface also has a similar feature that allows
you to specify the progress bar in the launcher.

![Linux Progress Bar][linux-progress-bar]

> NOTE: on Windows, each window can have its own progress bar, whereas on macOS
and Linux (Unity) there can be only one progress bar for the application.

----

All three cases are covered by the same API - the
[`setProgressBar()`][setprogressbar] method available on an instance of
`BrowserWindow`. To indicate your progress, call this method with a number
between `0` and `1`. For example, if you have a long-running task that is
currently at 63% towards completion, you would call it as
`setProgressBar(0.63)`.

> Setting the parameter to negative values (e.g. `-1`) will remove the progress
> bar, whereas setting it to values greater than `1` (e.g. `2`) will switch the
> progress bar to indeterminate mode. In this mode, a progress bar remains active
> but does not show an actual percentage. Use this mode for situations when you
> do not know how long an operation will take to complete. The indeterminate mode
> is a Windows-only feature.

Reviewer: ```suggestion
Setting the parameter to negative values (e.g. `-1`) will remove the progress
bar, whereas setting it to values greater than `1` (e.g. `2`) will switch the
progress bar to indeterminate mode (Windows-only -- it will clamp to 100% otherwise).
In this mode, a progress bar remains active but does not show an actual percentage.
Use this mode for situations when you do not know how long an operation
will take to complete.
```

Setting the parameter to negative values (e.g. `-1`) will remove the progress
bar, whereas setting it to values greater than `1` (e.g. `2`) will switch the

> progress bar to indeterminate mode (100% complete).

Reviewer: It seems like indeterminate mode is a Windows-only feature, and represents a progress bar that's active but doesn't show an actual percentage. We should make that clear in this line (Windows-only and doesn't represent 100% complete)

Ref: https://docs.microsoft.com/en-us/dotnet/api/system.windows.shell.taskbaritemprogressstate?view=netcore-3.1#System_Windows_Shell_TaskbarItemProgressState_Indeterminate
